### PR TITLE
Jazzy patch release 2024-12-23 

### PR DESCRIPTION
**Pull request by marcoag**
Jazzy patch release: https://discourse.ros.org/t/preparing-for-jazzy-sync-and-patch-release-2024-12-23/

CI:
* Linux [Build Status](https://ci.ros2.org/job/ci_linux/22313/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/16576/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/23105/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/1845/)

Packaging:
* Linux [Build Status](https://ci.ros2.org/job/ci_packaging_linux/565/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_packaging_linux-aarch64/111/)
* Windows [Build Status](https://ci.ros2.org/job/ci_packaging_windows/236/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_packaging_linux-rhel/71/)

### CODE REVIEW

**Comment by marcoag** [comment]
There seems to be a [cmake_minimum_required](https://ci.ros2.org/job/ci_windows/23105/cmake/category.-1518248354/category.-1518248354/category.-1518248354/) warning on all jobs coming from the iceoryx repo, I guess we can live with that...

File: ros2.repos [hunk 1]
```diff
@@ -230,23 +230,23 @@ repositories:
   ros2/geometry2:
     type: git
     url: https://github.com/ros2/geometry2.git
-    version: 0.36.4
+    version: 0.36.6
```
Comment (clalancette) [inline ros2.repos:233]: I see a tag for 0.36.7 in the repository, so we should update to that and merge in https://github.com/ros/rosdistro/pull/44052:

```suggestion
    version: 0.36.7
```

**Review by clalancette — CHANGES_REQUESTED** [review]
Besides what is in here, I think we should also probably do a release of `rosidl_typesupport_fastrtps`.  It doesn't have a huge amount of changes, but we may as well release it with everything else.

We should also do a bloom-release of `yaml_cpp_vendor`, as that is version 9.0.0 in https://github.com/ros/rosdistro/blob/6cf5283ecc1bbdbcd95562d243482088b7965765/jazzy/distribution.yaml#L9819

Finally, as for the iceoryx warning, it looks like that was fixed in iceoryx v2.0.6.  So I'm going to suggest that we update to that version here, and also do a bloom-release of that for Jazzy.

File: ros2.repos (hunk 1 above)
Comment (marcoag) [reply ros2.repos:233]: Sure. It seems recent so I missed it.

**Comment by marcoag** [comment]
Versions updated.

CI:
* Linux [Build Status](https://ci.ros2.org/job/ci_linux/22330/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/16593/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/23121/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/1861/)

Packaging:
* Linux [Build Status](https://ci.ros2.org/job/ci_packaging_linux/566/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_packaging_linux-aarch64/112/)
* Windows [Build Status](https://ci.ros2.org/job/ci_packaging_windows/237/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_packaging_linux-rhel/72/)

**Review by clalancette — APPROVED** [review]
Looks good to me, thank you for the updates!

**Pull request merged**